### ISSUE TITLE
request for ;pick to check field exp before removing gear and buffing

### ISSUE THREAD

**Issue by LuckyThiefDR** [rpherbig/dr-scripts#4339]
;pick doesn't check field exp until after removing equipment and buffing. This issue comes up when ;pick is part of a script list for T2 - you end up removing gear and buffing for no reason, and then immediately put all your stuff back on and end the script.

**Comment by Hiinky** [comment]
In my opinion, would be best fixed by using ;locksmithing instead of pick.  The exp check is built in so it never even starts ;pick if already locked.  ;Locksmithing will also handle daily trainers or consumable trainers like keepsake boxes from burgle.  Additional settings are minimal can be found on wiki: https://elanthipedia.play.net/Lich_script_repository#locksmithing

**Comment by Hiinky** [comment]
```
burgle_settings:   #<<<  this head already exists just add the before/after sections
  room: ####                                 # pick your own room, no guards should ever come here
  entry_type: lockpick                      # lockpick | rope | cycle
  use_lockpick_ring: true                   # if you use a worn lockpick stacker
  lockpick_container: lockpick ring         # name of your lockpick stacker
  max_search_count: 2                       # max number of total searches to do including retries
  retry: false                              # retry if you don't find loot.  default: false
  loot: drop                                # drop | keep | bin | pawn   default: keep (bin and pawn are not supported currently)
  loot_container: haversack                   # container to keep loot in while burgling
  safe_mode: true                           # stop searching and try to get out as quickly as possible when hearing footsteps.  Default: true
  room_blacklist:                           # array of rooms to ignore. Case sensitive.
  item_whitelist:                           # array of full item names (adj + noun) of items to keep no matter what.  Case Sensitive.  
    - keepsake box
  after:
    - go2 ####
    - locksmithing 
```
These are the burgle settings I use on my T2 characters to go from BNE directly to locks to minimize time/drain between the two actions plus collect keepsakes boxes.

Since I'm already pasted a big chunk... here's the basic locksmithing settings
```
## Locksmithing Settings ##

# use ;pick and pick settings first
pick_live_boxes: true

# second option to a daily use lockbox trainer like a training box or harvest bag
have_training_box: false
picking_lockbox: training box
picking_worn_lockbox: false

# IMPORTANT! Consumable boxes will be pulled from your burgle_settings: loot_container and should be kept separate from live boxes.
# list as many as you want - though it will try from the top down.
# THESE BOXES WILL BE DROPPED.  MAKE CERTAIN YOU ONLY LIST THROW-AWAY TRAINERS.
consumable_lockboxes:
# - keepsake box
# - liquor cabinet
```

**Comment by LuckyThiefDR** [comment]
That's fantastic, I sort of ignored ;locksmithing as an option so I'll give that shot.
Thanks

**Comment by Hiinky** [comment]
Just checking back to see if this solved your snag.  Please close the issue if you're we're all set.

**Comment by rpherbig** [comment]
Closed in #5001
